src/Inciter/Carrier.C

```diff
@@ -244,9 +244,57 @@ Carrier::vol()
 }
 
 void
+Carrier::comvol( const std::vector< std::size_t >& gid,
+                 const std::vector< tk::real >& V )
+// *****************************************************************************
+//  Receive nodal volumes on chare-boundaries
+//! \param[in] gid Global mesh node IDs at which we receive volume contributions
+//! \param[in] V Partial sums of nodal volume contributions to chare-boundary
+//!   nodes
+//! \details This function receives contributions to m_vol, which stores the
+//!   nodal volumes. While m_vol stores own contributions, m_volc collects the
+//!   neighbor chare contributions during communication. This way work on m_vol
+//!   and m_volc is overlapped. The two are combined in totalvol().
+// *****************************************************************************
+{
+  Assert( V.size() == gid.size(), "Size mismatch" );
+
+  for (std::size_t i=0; i<gid.size(); ++i) {
+    auto bid = tk::cref_find( m_bid, gid[i] );
+    Assert( bid < m_volc.size(), "Indexing out of bounds" );
+    m_volc[ bid ] += V[i];
+  }
+
+  if (++m_nvol == m_msum.size()) {
```

Comment:
What are the Charm++ rules regarding handling counters like this? Do we have a guarantee that entry invocations will always be atomic/serialized (unless otherwise flagged)? I know that in practice this is true, but if we don't have a guarantee then we are setting up a ticking time bomb of race conditions.

This is theoretically a problem elsewhere (which is why I assume it is probably fine), but this is the kind of thing we want to avoid in a fix branch unless we know it will be fine